web/src/components/storage/space-policy-utils.jsx

```diff
@@ -105,21 +108,64 @@ const SpacePolicySelector = ({ value, onChange = noop }) => {
   );
 };
 
-const SpacePolicyButton = ({ policy, onClick = noop }) => {
-  const text = (policy) => {
+const SpacePolicyButton = ({ policy, devices, onClick = noop }) => {
+  const text = (policy, num) => {
     switch (policy) {
       case "delete":
-        return _("deleting its current content");
+        return sprintf(
+          n_("deleting all content at the target disk", "deleting all content at the %d chosen disks", num),
+          num
+        );
       case "resize":
-        return _("shrinking existing partitions");
+        return sprintf(
+          n_(
+            "shrinking partitions at the target disk",
+            "shrinking partitions at the %d chosen disks",
+            num),
+          num
+        );
       case "keep":
         return _("without modifying any partition");
     }
     console.log("Unsupported value " + policy);
     return "error";
   };
 
-  return <Button variant="link" isInline onClick={onClick}>{text(policy)}</Button>;
+  const num = devices.length;
+
+  return <Button variant="link" isInline onClick={onClick}>{text(policy, num)}</Button>;
```

Comment:
No strong opinion to be honest, but when declaring and using functions like `text` here in the middle of the JSX I tend to use an internal React component instead. I.e., 

```js
const Text = ({ policy, num }) => { .... }
....
<Button ....>
  <Text policy={policy} num={devices.length} />
</Button>
```

What is more, you could even go further and remove these props since it's an internal component.

But it's just another way to do it, to ending up having a JSX block with less interpolation.